### PR TITLE
Add enabled flag to Item

### LINKED ISSUE #56: Add enabled flag to Item
Update bindings to add enabled flag as per [#1175](https://github.com/PixarAnimationStudios/OpenTimelineIO/pull/1175)

### PR DESCRIPTION
**Pull request by KarthikRIyer**
Closes #56 

This PR adds:
- The enabled flag to Item.
- Proposes to add and further write tests in the BDD (Behaviour Driven Development) format.

The idea of BDD testing is to write tests in a simple user/business friendly manner. The tests become a documentation in themselves. For example:

If we want to write a simple test to see if disabling an `Item` works...

The BDD specification would be like:

```
given: "An item is created"
when: "Item is disabled"
then: "Item does not contribute to the composition"
```

And the code would be like: 

```groovy
given: "An item is created"
def item = new Item()
verify {
    item.enabled() == true
}

when: "Item is disabled"
item.setEnabled(false)

then: "Item does not contribute to the composition"
verifyAll {
        item.isEnabled() == false
}
```

The PR has the same tests, both in the unit testing format and BDD format, for comparison.

I've used [spock](https://spockframework.org/), a groovy BDD testing framework. It might be a lot of effort in the beginning but once you get used to it, it becomes simpler and clearer. It'll also be helpful to community members new to the project. The current tests might not be very clear to them. 
Spock also has an extension [spock-reports](https://github.com/renatoathaydes/spock-reports) that generates nice HTML test reports. But I haven't been able to get it working with the bindings yet.

Any thoughts on taking this forward? Or should we continue writing unit test as we do right now?

If we do take it forward, we might also want to adopt this for the python bindings. There are plenty of frameworks available, but in particular `pytest-bdd` is compatible with pytest. ([Framework comparison](https://automationpanda.com/2019/04/02/python-bdd-framework-comparison/))

@reinecke @meshula @jminor @ssteinbach

### CODE REVIEW

**Comment by KarthikRIyer** [comment]
Spock reports now works!

image
image

**Comment by KarthikRIyer** [comment]
Hi @reinecke, could you please take a look at this too

File: src/main/java/io/opentimeline/opentimelineio/Item.java [hunk 1]
```diff
@@ -39,7 +55,8 @@ public Item(
                 sourceRange,
                 metadata,
                 effects,
-                markers);
+                markers,
+                true);
     }
```
Comment (reinecke) [inline src/main/java/io/opentimeline/opentimelineio/Item.java:60]: Should we mark this constructor with an `@Deprecated` to encourage people toward the constructor that best matches the underlying implementation?

**Review by reinecke — CHANGES_REQUESTED** [review]
In terms of `isEnabled` - one small question on deprecating the old constructor.

On BDD testing:

I support migrating to spock tests for our Java implementation, however I would prefer that we do that its own separate PR.

In terms of BDD testing in python, I think that choice is entangled with some language-specific details (like which frameworks we use, what the current best practice in the community is), so we should have that discussion on an issue in the [main repo](https://github.com/PixarAnimationStudios/OpenTimelineIO).

File: src/main/java/io/opentimeline/opentimelineio/Item.java (hunk 1 above)
Comment (KarthikRIyer) [reply src/main/java/io/opentimeline/opentimelineio/Item.java:60]: Yes. I'll make the change

**Comment by KarthikRIyer** [comment]
@reinecke this PR only has the wrapper for the new flag now. I think this is good to merge.

**Review by reinecke — APPROVED** [review]
Thanks for taking care of this!

**Pull request merged**